flink-runtime/src/main/java/org/apache/flink/runtime/metrics/groups/InternalSinkWriterMetricGroup.java

```diff
@@ -35,12 +35,20 @@
         implements SinkWriterMetricGroup {
 
     private final Counter numRecordsOutErrors;
+    private final Counter numRecordsWritten;
+    private final Counter numBytesWritten;
     private final OperatorIOMetricGroup operatorIOMetricGroup;
 
     private InternalSinkWriterMetricGroup(
             MetricGroup parentMetricGroup, OperatorIOMetricGroup operatorIOMetricGroup) {
         super(parentMetricGroup);
         numRecordsOutErrors = parentMetricGroup.counter(MetricNames.NUM_RECORDS_OUT_ERRORS);
+        numRecordsWritten =
+                parentMetricGroup.counter(
+                        org.apache.flink.runtime.metrics.MetricNames.IO_NUM_RECORDS_OUT);
```

Comment:
This doesn't really work. You are registering a second counter with the same name of one that already exists, returning you a dummy counter which is never exported. You could just as well create a plain SimpleCounter if that is the intention.

Reply:
> Second question, What do you mean "never exported"? Is there anything else should be done to make it be exported? Do we have tech guide somewhere?

With that I mean that the metric will never be passed to a reporter. You can't work your way around that, beyond solving the issue of registering 2 metrics with the same name on the same group.

Reply:
Thanks for the feedback. Comments have been addressed.